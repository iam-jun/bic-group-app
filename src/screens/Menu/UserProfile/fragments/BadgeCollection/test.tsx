import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
