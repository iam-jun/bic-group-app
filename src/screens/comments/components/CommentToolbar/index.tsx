import React, { FC } from 'react';
import {
  StyleProp, StyleSheet, View, ViewStyle,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import Icon from '~/baseComponents/Icon';
import Button from '~/beinComponents/Button';
import spacing from '~/theme/spacing';

export interface CommentToolbarProps {
  style?: StyleProp<ViewStyle>;
  disableImageOption?: boolean;
  disableGifOption?: boolean;

  onSelectEmoij?: () => void;
  onSelectImage?: () => void;
  onSelectGif?: () => void;
}

const CommentToolbar: FC<CommentToolbarProps> = ({
  style,
  disableImageOption,
  disableGifOption,
  onSelectEmoij,
  onSelectImage,
  onSelectGif,
}: CommentToolbarProps) => {
  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);
  const { colors } = theme;

  return (
    <View style={[styles.container, style]}>
      {!!onSelectEmoij
      && (
      <Button
        testID="comment_toolbar.btn_icon"
        style={styles.button}
        onPress={onSelectEmoij}
      >
        <Icon
          icon="iconReact"
        />
      </Button>
      )}
      {!!onSelectImage && (
        <Button
          testID="comment_toolbar.btn_image"
          disabled={disableImageOption}
          style={styles.button}
          onPress={onSelectImage}
        >
          <Icon
            icon="Image"
            size={18}
            tintColor={disableImageOption ? colors.gray20 : colors.neutral40}
          />
        </Button>
      )}
      {!!onSelectGif && (
        <Button
          testID="comment_toolbar.btn_gif"
          disabled={disableGifOption}
          style={styles.button}
          onPress={onSelectGif}
        >
          <Icon
            icon="iconAddGif"
            tintColor={disableGifOption ? colors.gray20 : colors.neutral40}
          />
        </Button>
      )}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      height: 56,
      width: '100%',
      flexDirection: 'row',
      alignItems: 'center',
      paddingHorizontal: spacing.padding.large,
      backgroundColor: colors.white,
      borderTopWidth: 1,
      borderColor: colors.neutral5,
      zIndex: 3,
    },
    button: {
      padding: spacing.padding.xSmall,
    },
  });
};

export default CommentToolbar;
