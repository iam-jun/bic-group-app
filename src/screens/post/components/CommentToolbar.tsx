import React, { FC } from 'react';
import {
  StyleProp, StyleSheet, View, ViewStyle,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import Icon from '~/baseComponents/Icon';
import Button from '~/beinComponents/Button';
import spacing from '~/theme/spacing';

interface CommentToolbarProps {
  style?: StyleProp<ViewStyle>;
  onSelectImage?: () => void;
  onSelectFile?: () => void;
  onSelectGif?: () => void;
  onSelectVideo?: () => void;
}

const CommentToolbar: FC<CommentToolbarProps> = ({
  style,
  onSelectImage,
  onSelectFile,
  onSelectGif,
  onSelectVideo,
}: CommentToolbarProps) => {
  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);

  if (!onSelectImage && !onSelectFile && !onSelectGif && !onSelectVideo) {
    return null;
  }

  return (
    <View style={[styles.container, style]}>
      {!!onSelectImage && (
        <Button
          testID="comment_toolbar.btn_image"
          style={styles.button}
          onPress={onSelectImage}
        >
          <Icon icon="Image" />
        </Button>
      )}
      {!!onSelectFile && (
        <Button
          testID="comment_toolbar.btn_file"
          style={styles.button}
          onPress={onSelectFile}
        >
          <Icon icon="Link" />
        </Button>
      )}
      {!!onSelectGif && (
        <Button
          testID="comment_toolbar.btn_gif"
          style={styles.button}
          onPress={onSelectGif}
        >
          <Icon icon="iconAddGif" />
        </Button>
      )}
      {!!onSelectVideo && (
        <Button
          testID="comment_toolbar.btn_video"
          style={styles.button}
          onPress={onSelectVideo}
        >
          <Icon icon="Video" />
        </Button>
      )}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      height: 56,
      width: '100%',
      flexDirection: 'row',
      alignItems: 'center',
      paddingHorizontal: spacing.padding.large,
      backgroundColor: colors.white,
      borderTopWidth: 1,
      borderColor: colors.gray40,
    },
    button: {
      padding: 2,
      marginRight: spacing.margin.large,
    },
  });
};

export default CommentToolbar;
