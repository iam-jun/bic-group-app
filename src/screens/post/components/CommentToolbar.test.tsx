import * as React from 'react';
import CommentToolbar from './CommentToolbar';
import { fireEvent, renderWithRedux, cleanup } from '~/test/testUtils';

afterEach(cleanup);

describe('CommentToolbar component', () => {
  it('renders correctly', () => {
    const onSelectImage = jest.fn();
    const onSelectFile = jest.fn();
    const onSelectGif = jest.fn();
    const onSelectVideo = jest.fn();
    const rendered = renderWithRedux(
      <CommentToolbar
        onSelectImage={onSelectImage}
        onSelectFile={onSelectFile}
        onSelectGif={onSelectGif}
        onSelectVideo={onSelectVideo}
      />,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('renders correctly null', () => {
    const rendered = renderWithRedux(<CommentToolbar />);
    expect(rendered).toMatchSnapshot();
  });

  it('should call prop onSelectImage', () => {
    const onSelect = jest.fn();
    const { getByTestId } = renderWithRedux(
      <CommentToolbar onSelectImage={onSelect} />,
    );
    const btn = getByTestId('comment_toolbar.btn_image');
    expect(btn).toBeDefined();
    fireEvent.press(btn);
    expect(onSelect).toBeCalled();
  });

  it('should call prop onSelectFile', () => {
    const onSelect = jest.fn();
    const { getByTestId } = renderWithRedux(
      <CommentToolbar onSelectFile={onSelect} />,
    );
    const btn = getByTestId('comment_toolbar.btn_file');
    expect(btn).toBeDefined();
    fireEvent.press(btn);
    expect(onSelect).toBeCalled();
  });

  it('should call prop onSelectGif', () => {
    const onSelect = jest.fn();
    const { getByTestId } = renderWithRedux(
      <CommentToolbar onSelectGif={onSelect} />,
    );
    const btn = getByTestId('comment_toolbar.btn_gif');
    expect(btn).toBeDefined();
    fireEvent.press(btn);
    expect(onSelect).toBeCalled();
  });

  it('should call prop onSelectVideo', () => {
    const onSelect = jest.fn();
    const { getByTestId } = renderWithRedux(
      <CommentToolbar onSelectVideo={onSelect} />,
    );
    const btn = getByTestId('comment_toolbar.btn_video');
    expect(btn).toBeDefined();
    fireEvent.press(btn);
    expect(onSelect).toBeCalled();
  });
});
