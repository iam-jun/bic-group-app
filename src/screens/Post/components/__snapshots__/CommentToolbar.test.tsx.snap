// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentToolbar component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#A4A9B7",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "height": 56,
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 16,
        "opacity": 1,
        "padding": 2,
      }
    }
    testID="comment_toolbar.btn_image"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 16,
        "opacity": 1,
        "padding": 2,
      }
    }
    testID="comment_toolbar.btn_file"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 16,
        "opacity": 1,
        "padding": 2,
      }
    }
    testID="comment_toolbar.btn_gif"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View>
              <View
                onLayout={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "position": "relative",
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#f9f9f9",
                        "height": 20,
                        "justifyContent": "center",
                        "width": 20,
                      }
                    }
                    testID="Image__placeholder"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                >
                  <FastImageView
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    resizeMode="cover"
                    size={20}
                    source={null}
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginRight": 16,
        "opacity": 1,
        "padding": 2,
      }
    }
    testID="comment_toolbar.btn_video"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CommentToolbar component renders correctly null 1`] = `null`;
